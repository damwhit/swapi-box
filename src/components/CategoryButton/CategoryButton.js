import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';

function CategoryButton({ category, onClick }) {
  const linkPath = `/${category}`;
  return (
    <NavLink to={linkPath} className="link--category">
      <button 
        className="btn--category"
        onClick={onClick}
        type="button"
      >
        {category}
      </button>
    </NavLink>
  );
}

CategoryButton.propTypes = {
  category: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default CategoryButton;
