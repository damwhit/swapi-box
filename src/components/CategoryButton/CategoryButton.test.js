import React from 'react';
import { shallow } from 'enzyme';
import CategoryButton from './CategoryButton';

describe('CategoryButton', () => {
  const onClickMock = jest.fn();
  const wrapper = shallow(
    <CategoryButton
      category='people'
      onClick={onClickMock}
    />
  );
  

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the onClick prop when clicked', () => {
    const button = wrapper.find('.btn--category');
    button.simulate('click');
    expect(onClickMock).toHaveBeenCalled();
  });
});
