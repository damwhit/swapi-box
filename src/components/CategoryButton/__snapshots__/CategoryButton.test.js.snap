// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryButton
    category="people"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": <button
        className="btn--category"
        onClick={[MockFunction]}
        type="button"
      >
        people
      </button>,
      "className": "link--category",
      "to": "/people",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "people",
        "className": "btn--category",
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": "people",
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": <button
          className="btn--category"
          onClick={[MockFunction]}
          type="button"
        >
          people
        </button>,
        "className": "link--category",
        "to": "/people",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
          "className": "btn--category",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "people",
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
